base-theme/assets/js/sentry.js

```diff
@@ -1,10 +1,14 @@
 /* eslint-disable no-console */
 import * as Sentry from "@sentry/browser"
+import { isLocalServerRunning } from "./utils"
 
 export const initSentry = () => {
-  Sentry.init({
-    dsn:
-      "https://eee58f41dda54d2b814296e12dced4b7@o48788.ingest.sentry.io/5304953"
-  })
-  return Sentry
+  if (!isLocalServerRunning()) {
+    Sentry.init({
+      dsn:
+        "https://eee58f41dda54d2b814296e12dced4b7@o48788.ingest.sentry.io/5304953"
+    })
```

Comment:
Sentry has a [`beforeSend`](https://docs.sentry.io/platforms/javascript/configuration/filtering/#using-platformidentifier-namebefore-send-) configuration option that can be used to discard the event:
> ...  If you return null, the event will be discarded.

IMO, it would be a little better to do
```ts
beforeSend: event => isLocalServerRunning() ? null : event
```
because that way `window.Sentry` will be a sentry object in all environments—we won't need modifications to `base-theme/assets/index.ts` or  `base-theme/layouts/partials/sentry_capture_message.html`

Reply:
Nice suggestion
Thanks! @ChristopherChudzicki